src/main/java/io/logz/jmx2graphite/MetricsPipeline.java

```diff
@@ -31,19 +35,36 @@ public MetricsPipeline(Jmx2GraphiteConfiguration conf, MBeanClient client) {
                                                  conf.getGraphiteProtocol());
         this.client = client;
         this.pollingIntervalSeconds = conf.getMetricsPollingIntervalInSeconds();
+        this.beansWhiteListPattern = conf.getWhiteListPattern();
+        this.beansBlackListPattern = conf.getBlackListPattern();
+
+    }
+
+    public List<MetricBean> getFilteredBeans(List<MetricBean> beans) {
+        List<MetricBean> filteredBeans = beans.stream()
+                .filter(bean -> beansWhiteListPattern.matcher(bean.getName()).find())
+                .collect(Collectors.toList());
+
+        filteredBeans.removeAll(beans.stream()
+                .filter((bean -> beansBlackListPattern.matcher(bean.getName()).find()))
+                .collect(Collectors.toList()));
+        return filteredBeans;
     }
 
     private List<MetricValue> poll() {
         try {
             long pollingWindowStartSeconds = getPollingWindowStartSeconds();
             Stopwatch sw = Stopwatch.createStarted();
             List<MetricBean> beans = client.getBeans();
+
             logger.info("Found {} metric beans. Time = {}ms, for {}", beans.size(),
                     sw.stop().elapsed(TimeUnit.MILLISECONDS),
                     new Date(TimeUnit.SECONDS.toMillis(pollingWindowStartSeconds)));
+            List<MetricBean> filteredBeans = getFilteredBeans(beans);
```

Comment:
To make it a bit more readable: `getFilteredBeans(beans)` --> `filterWhiteBlackList(beans)`